src/controllers/posts.ts

```diff
@@ -0,0 +1,121 @@
+import { Request, Response, NextFunction } from 'express';
+import { StatusCodes } from 'http-status-codes';
+import type { TokenPayload } from '../factories/authentication/authentication-client.types';
+import { type Pagination, Sorting } from '../interfaces/pagination';
+import type { ServiceDependencies } from '../services';
+
+export type PostHandlers = keyof ReturnType<typeof createPostsController>;
+
+export const createPostsController = ({ postService }: ServiceDependencies) => {
+  const addPost = async (req: Request, res: Response, next: NextFunction) => {
+    const posts = await postService.addPost((res.locals.user as TokenPayload).userId, req.body);
+
+    return res.status(StatusCodes.OK).send({
+      code: StatusCodes.OK,
+      message: 'Post has been successfully added.',
+      data: posts,
```

Comment:
Are we getting here all user posts after adding new one? If so, what for?

Reply:
Yes, all posts including the new post, so you can replace the current state of posts. I can change it if you want.

Reply:
So please remove it. By design we can add new post from any page not only from our user page, so in most cases it would be just unnecessary data in response